core/backend/Process/LegacyHandler/CreatePortalUserActionHandler.php

```diff
@@ -0,0 +1,211 @@
+        }
+
+        $options = $process->getOptions();
+
+        if (empty($options['id'])) {
+            throw new InvalidArgumentException(self::MSG_OPTIONS_NOT_FOUND);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     * @throws \JsonException
+     */
+    public function run(Process $process)
+    {
+        $this->init();
+
+        $options = $process->getOptions();
+
+        global $sugar_config, $mod_strings;
+        $contact = \BeanFactory::getBean('Contacts', $options['id']);
+
+        if(!array_key_exists("aop", $sugar_config)){
+            $this->logger->warning('LBL_ERROR_AOP_NOT_CONFIGURED');
+            $process->setMessages(['LBL_ERROR_AOP_NOT_CONFIGURED']);
+            $process->setStatus('error');
+            return;
+        }
+        if(empty($sugar_config['aop']['enable_portal'])){
+            $this->logger->warning('LBL_ERROR_PORTAL_NOT_ENABLED');
+            $process->setMessages(['LBL_ERROR_PORTAL_NOT_ENABLED']);
+            $process->setStatus('error');
+            return;
+        }
+        if(empty($sugar_config['aop']['enable_aop'])){
+            $this->logger->warning('LBL_ERROR_AOP_NOT_ENABLED');
+            $process->setMessages(['LBL_ERROR_AOP_NOT_ENABLED']);
+            $process->setStatus('error');
+            return;
+        }
+        if(empty($sugar_config['aop']['joomla_url'])){
+            $this->logger->warning('LBL_ERROR_JOOMLA_URL_MISSING');
+            $process->setMessages(['LBL_ERROR_JOOMLA_URL_MISSING']);
+            $process->setStatus('error');
+            return;
+        }
+
+        if ($contact->id && $contact->email1) {
+            $portalURL = $sugar_config['aop']['joomla_url'];
+            $wbsv = file_get_contents($portalURL.'/index.php?option=com_advancedopenportal&task=create&sug='.$contact->id);
+            if($wbsv === false) {
+                $this->logger->error($mod_strings['LBL_FAILED_TO_CONNECT_JOOMLA']);
+                $process->setStatus('error');
+                $process->setMessages(['LBL_CREATE_PORTAL_USER_FAILED']);
+                return;
+            }
+            $res = json_decode($wbsv, false, JSON_THROW_ON_ERROR, JSON_THROW_ON_ERROR);
+            if (!$res->success) {
+                $msg = $res->error ?: $mod_strings['LBL_CREATE_PORTAL_USER_FAILED'];
+                $this->logger->error($msg);
+                $this->logger->error('JSON error: ' . json_last_error_msg());
+                $process->setStatus('error');
+                $process->setMessages([$msg]);
+            } else {
+                $process->setMessages(['LBL_CREATE_PORTAL_USER_SUCCESS']);
+                $process->setStatus('success');
+                $process->setData(['reload' => true]);
+            }
+        } else {
+            $this->logger->error($mod_strings['LBL_ERROR_CONTACT_ID_OR_EMAIL_EMPTY']);
+            $process->setMessages(['LBL_ERROR_CONTACT_ID_OR_EMAIL_EMPTY']);
+            $process->setStatus('error');
+        }
```

Comment:
Hi @eojedapilchik in order to make this easier to read, maybe this could be moved to an early return

Reply:
refactored